### PR TITLE
Improve delta chat controller structure

### PR DESCRIPTION
**Pull request by Simon-Laux**
this brings back intellisense and typeinformations.
progress:
- [X] './autocrypt'
- [X] './backup'
- [X] './chatlist'
- [x] './chatmethods'
- [x] './locations'
- [x] './login'
- [X] './messagelist'
- [X] './settings'
- [X] './stickers'

my steps are: 
1. transform to class
2. check whether to expose them
3. change call signature where they are used
4. make sure that they don't access stuff on this that is under this._controller

### CODE REVIEW

File: src/main/deltachat/index.js [hunk 1]
```diff
@@ -62,16 +66,45 @@ class DeltaChatController extends EventEmitter {
     logCoreEv.debug(event, data1, data2)
   }
 
+  __resolveNestedMethod (self, methodName) {
+    let previous = self
+    let scope
+    for (const key of methodName.split('.')) {
```
Comment (Jikstra) [inline src/main/deltachat/index.js]: How about making this simpler and only allowing a 2 level nesting? We should never have more than namespace.method

File: src/main/deltachat/index.js [hunk 2]
```diff
@@ -42,46 +42,59 @@ class DeltaChatController extends EventEmitter {
...
+      autocrypt: new DCAutocrypt(this),
+      backup: new DCBackup(this),
+      chatList: new DCChatList(this),
+      contacts: new DCContacts(this),
+      chat: new DCChat(this),
+      locations: new DCLocations(this),
+      loginController: new DCLoginController(this),
+      messageList: new DCMessageList(this),
+      settings: new DCSettings(this),
+      stickers: new DCStickers(this)
```
Comment (Jikstra) [inline src/main/deltachat/index.js]: Ah here we go.

File: src/main/deltachat/index.js [hunk 3]
```diff
@@ -22,10 +27,17 @@ class DeltaChatController extends EventEmitter {
...
     if (!saved) throw new Error('Saved settings are a required argument to DeltaChatController')
     this.loadSplitOuts()
+    /**
+     * @type {DeltaChat}
+     */
+    this._dc = undefined
+  }
+
+  get autocrypt () {
+    return new DCAutocrypt(this._dc)
```
[the remark this answers is not in the document]
Comment (Simon-Laux) [reply src/main/deltachat/index.js]: shouldn't have been a big issue anyway (just instancing a class and assigning 2 vars

File: src/main/deltachat/index.js (hunk 1 above)
Comment (Simon-Laux) [reply src/main/deltachat/index.js]: I also thought about this, I can do that.

Comment (Simon-Laux) [reply src/main/deltachat/index.js]:
```
__resolveNestedMethod (self, methodName) {
    const parts = methodName.split('.')
    if (parts.length > 2) {
      const message = 'Resolving of nested method name failed: Too many parts, only two allowed: ' + methodName
      log.error(message)
      throw new Error(message)
    }
    const scope = self[parts[0]]
    if (typeof scope === 'undefined') {
      const message = 'Resolving of nested method name failed: ' + methodName
      log.error(message)
      throw new Error(message)
    }
    const method = scope[parts[1]]
    if (typeof method !== 'function') {
      const message = '(nested) Method is not of type function: ' + methodName
      log.error(message)
      throw new Error(message)
    }
    return method.bind(scope)
  }
```
Doesn't look so much simpler, does it?

**Pull request merged**